sway/criteria.c

```diff
@@ -370,46 +370,87 @@ static char *get_focused_prop(enum criteria_token token) {
 	struct sway_seat *seat = input_manager_current_seat();
 	struct sway_container *focus = seat_get_focused_container(seat);
 
-	if (!focus || !focus->view) {
-		return NULL;
-	}
-	struct sway_view *view = focus->view;
+	struct sway_view *view = focus ? focus->view : NULL;
 	const char *value = NULL;
 
+	// For any of these tokens that accept __focused__, they should still
+	// return a valid value when there is no focus. The value should not
+	// match any possible view. Since most of these could be empty strings,
+	// that is not an option. This uses a backspace character, which is very
+	// unlikely to be in any of these strings. For T_CON_ID, the string should
+	// contain an unsigned integer so \b does not make sense. Instead 1, which
+	// is the id for root is used (0 may equate to an empty criteria if
+	// T_CON_ID is the only token).
+	//
+	// Additionally, if there is a focused view and it does not have the
+	// property set, \b or 1 will also be used to ensure a valid value is
+	// returned.
+
 	switch (token) {
 	case T_APP_ID:
-		value = view_get_app_id(view);
+		if (view) {
+			value = view_get_app_id(view);
+		}
+		if (!value) {
+			value = "\b";
```

Comment:
I'm not a fan of this `\b` thing. I'd prefer a separate `bool` value.

Reply:
I'll rework this and make it more elegant

Reply:
Should be more elegant now